packages/echo/src/client.ts

```diff
@@ -626,11 +633,16 @@ export class Echo {
   }
 
   private createExecutionInputs(event: IEvent, workflow: DiscoverWorkflowOutput): Record<string, unknown> {
-    const executionData = event.data;
+    let payload = event.data;
+    if (event.action === 'preview') {
+      const mockResult = this.mock(workflow.data.schema);
+
+      payload = Object.assign(mockResult, payload);
```

Comment:
Wonder if we should do something more smart here like recursive replace?

Reply:
I think we should go one way or the other:
* No replacement at all
* Recursive replacement

If we're halfway, it's likely we'll end up the unexpected results when it comes to nested objects.

Reply:
Any suggestions of how to achieve this? Can install something like lodash deep merge perhaps?

Reply:
I have a strong preference to keep this library very lightweight for fast execution on serverless/low-compute environments. Adding lodash feels like a bit of an overkill for now. I added a suggested below.